libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx

```diff
@@ -0,0 +1,85 @@
+import { Controller, useFormContext } from 'react-hook-form'
+import { ApplicationEntity } from '@console/shared/interfaces'
+import { BlockContent, Button, ButtonSize, ButtonStyle, HelpSection, Icon, InputToggle } from '@console/shared/ui'
+import { IconEnum } from '@console/shared/enums'
+
+export interface PageSettingsPreviewEnvironmentsProps {
+  onSubmit: () => void
+  watchEnvPreview: boolean
+  applications?: ApplicationEntity[]
+}
+
+export function PageSettingsPreviewEnvironments(props: PageSettingsPreviewEnvironmentsProps) {
+  const { onSubmit, applications, watchEnvPreview } = props
+
+  const { control, formState } = useFormContext()
+
+  return (
+    <div className="flex flex-col justify-between w-full">
+      <div className="p-8">
+        <form onSubmit={onSubmit}>
+          <BlockContent title="General">
+            <Controller
+              name="auto_preview"
+              control={control}
+              render={({ field }) => (
+                <InputToggle
+                  value={field.value}
+                  onChange={field.onChange}
+                  title="Activate preview environment for all applications"
+                  description="Automatically create a preview environment when a merge request is submitted on one of your applications. Your environment will be cloned with the application synchronised on the branch waiting to be merged."
```

Comment:
Automatically create a preview environment when a merge/pull request is submitted on one of your applications.